### PR TITLE
feat(PPDSC-2510): add dismissOnBlur prop to Popover

### PR DESCRIPTION
**Pull request by mutebg**
PPDSC-2510

**What**

1. Background - it's required by Dow Jones
2. What did you do - added a new prop `dismissOnBlur`, when is set to true the popover will automatically close.
3. What does the reviewers should expect

**I have done:**
- [x] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [x] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by newskit-engineering** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2510-dismiss-popover-on-focus-away
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2510-dismiss-popover-on-focus-away/storybook

**Review by JohnTParsons — APPROVED** [review]

File: site/pages/components/popover.tsx [hunk 1]
```diff
@@ -595,6 +595,22 @@ const PopoverComponent = (layoutProps: LayoutProps) => (
...
+              description:
+                'If true, the Popover can be closed by focusing outside of the popover.',
+              required: undefined,
+            },
+            {
+              name: 'disableFocusManagement',
+              type: 'boolean',
+              default: 'false',
+              description:
+                'If true, the Popover will not manage focus focus for you which means when open will not move the focus to the popover and back to the trigger when closed',
```
Comment (mstuartf) [inline site/pages/components/popover.tsx:611]: typo: `focus focus`

File: src/popover/__tests__/popover.test.tsx [hunk 2]
```diff
@@ -385,6 +385,34 @@ describe('Popover', () => {
       await applyAsyncStyling();
       expect(queryByRole('dialog')).not.toBeInTheDocument();
     });
+
+    test('does close on focus outside', async () => {
```
Comment (mstuartf) [inline src/popover/__tests__/popover.test.tsx:389]: ...when `dismissOnBlur` is true.

**Review by mstuartf — COMMENTED** [review]
a few small comments only, otherwise looks good 👍

**Review by mstuartf — APPROVED** [review]

**Review by JohnTParsons — APPROVED** [review]

**Pull request merged**